### ISSUE TITLE
Ban property

### ISSUE THREAD

**Issue by andy-ms** [palantir/tslint#2791]
### Bug Report

- __TSLint version__: 5.2.0
- __TypeScript version__: 2.4.0-dev.20170519
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
[-1, 2, -3].map(Math.abs);
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "ban": [
            true,
            ["Math", "abs"]
        ]
    }
}
```

#### Actual behavior

No error.

#### Expected behavior

A lint failure.

This would be a breaking change, but it makes sense to me. Maybe put it under an option until 6.0?

**Comment by vsiao** [comment]
How come `ban` is restricted to call expressions? Can we just remove this line?
 https://github.com/palantir/tslint/blob/master/src/rules/banRule.ts#L149

**Comment by ryanatkn** [comment]
Another use case - I want to ban usage of `length` and `name` on the DOM window global. It's a rare gotcha but it's gotten me multiple times after refactoring.

```ts
console.log('surprise, these are defined in lib.dom.d.ts', length, name);
```

The following TypeScript-based solutions don't work: (because `Subsequent variable declarations must have the same type.`)

```ts
declare var length: never;
// nor
interface Window {
  length: never;
}
```

**Comment by jdom** [comment]
@ryanatkn I believe your case would be covered by #3824.

I'm looking for banning those global variables, but also usages of `Function.length`. Could this rule be expanded to cover property accesses too?

**Comment by JoshuaKGoldberg** [comment]
Removing the `Type: Breaking Change` label per #4811. Now accepting PRs!
